### PR TITLE
Bcarvalheira/get deleted

### PR DESCRIPTION
**Pull request by carvalhe**
Add getDeletedSecret from keyvault api

### CODE REVIEW

File: pkg/store/store.go [hunk 1]
```diff
@@ -85,6 +85,45 @@ func (s *MsiKeyVaultStore) GetCredentialsObject(ctx context.Context, secretName
...
+	}
+
+	var credentialsObject dataplane.CredentialsObject
+	if err := credentialsObject.UnmarshalJSON([]byte(*response.Value)); err != nil {
+		return nil, err
+	}
+
+	secretProperties := SecretProperties{
+		Name:      secretName,
+		Enabled:   true, // Default to true
```
Comment (niontive) [inline pkg/store/store.go:106]: If the secret is deleted, it shouldn't be enabled.

Comment (niontive) [reply pkg/store/store.go:106]: Also, see attributes section: https://github.com/Azure/azure-sdk-for-go/blob/7664dd24aeda13ec5b7af4531047815c4682ca36/sdk/security/keyvault/azsecrets/models.go#L21

This has Enabled, expires, etc.

File: pkg/store/store.go [hunk 2]
```diff
@@ -85,6 +85,45 @@ func (s *MsiKeyVaultStore) GetCredentialsObject(ctx context.Context, secretName
 	return &SecretResponse{CredentialsObject: credentialsObject, Properties: secretProperties}, nil
 }
 
+// Get a deleted credentials object from the key vault using the specified secret name.
+func (s *MsiKeyVaultStore) GetDeletedCredentialsObject(ctx context.Context, secretName string) (*SecretResponse, error) {
```
Comment (niontive) [inline pkg/store/store.go:89]: Let's create a separate struct called `DeletedSecretResponse` to make it clear this is a deleted secret response.

Comment (carvalhe) [reply pkg/store/store.go]: Added both DeletedSecretResponse and DeletedSecretProperties

**Review by niontive — APPROVED** [review]

**Pull request merged**